docs/developer/payments/lifecycle.mdx

```diff
@@ -173,27 +259,152 @@ it may still remain unpaid if the pending amounts fail to process.
+### Order of events
+
+New events can be reported to Saleor in non-subsequent order. This means that some old event might arrive later than a new one. For example:
+- on 10 AM Saleor receives `CHARGE_SUCCESS` event from payment app that was processed at 9:50 AM by payment provider
+- on 10:20 AM Saleor receives `AUTHORIZATION_SUCCESS` event, that was process at 9:30 AM by payment provider.
+
+This will result in a following list of events:
+- `AUTHORIZATION_SUCCESS` with `date: 9:30 AM`
+- `CHARGE_SUCCESS` with `date: 9:50 AM`
+
+### Reporting new events
+
+#### Automatic events
+
+Events are created automatically by Saleor when calling `transactionInitialize` mutation. It creates either:
+- `AUTHORIZATION_REQUEST` when `action` in mutation input or `channel.transactionFlowStrategy` is `AUTHORIZATION`
+- or `CHARGE_REQUEST` when `action` in mutation input or `channel.transactionFlowStrategy` is `CHARGE`
+
+Events are also created when requesting action in Saleor (refund, charge, cancel) via [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action.mdx) mutation.
+
+Such events don't have initially any `pspReference`, it's filled in once Saleor receives a response from payment app.
+
+Saleor also creates automatically `_FAILURE` events when an exception happens:
+- when handling `transactionEventReport` mutation
+- when handling incorrect response from [sync webhook sent to payment app](/developer/extending/webhooks/synchronous-events/transaction.mdx) (e.g. app didn't respond in 20s, returned JSON is invalid, etc.)
+
+Such events don't have `pspReference` and include a `message` that explains the error.
+
+#### App events
+
+Events are also created by a payment app that processes the payment:
+
+- When payment app sends response to any of the [Transaction synchronous events](/developer/extending/webhooks/synchronous-events/transaction.mdx)
+
+:::info
+Instead of creating new event, Saleor will update existing event's missing `pspReference` field when:
+- Event without `pspReference` already exists, because it was automatically created by Saleor after calling `transactionInitialize` or `transactionProcess` mutation
+- Response from `TRANSACTION_INITIALIZE_SESSION` or `TRANSACTION_PROCESS_SESSION` has the same `type` as that initial event
+
+When updating event `createdAt` field won't be changed.
+:::
+
+:::note
+It can look like this:
+1. User requests payment by calling `transactionInitialize` mutation - Saleor immediately created new `CHARGE_REQUESTED` event
+2. App sends response from `TRANSACTION_INITIALIZE_SESSION` webhook with `type`: `CHARGE_ACTION_REQUIRED`
+3. User completes the action and calls `transactionProcess`
+4. App sends response from `TRANSACTION_PROCESS_SESSION` webhook with the same type as first event - `CHARGE_REQUESTED`
+5. Saleor instead of creating new event, updates existing event `CHARGE_REQUESTED` with received `pspReference`
+
+Note that this will result in events order:
```

Comment:
**Question:** you mean here order of events on checkout / order? If so maybe 

`Note that this will result in events ordered as follows`?

Reply:
Yup, I give an example situation and what what will be the order of `TransactionEvent`s in Saleor.